website/content/docs/api-gateway/upgrades.mdx

```diff
@@ -9,6 +9,151 @@ description: >-
+
+### Requirements
+
+Ensure that the following requirements are met prior to upgrading:
+
+- Consul API Gateway should be running version v0.3.0.
+
+### Procedure
+
+1. Proceed with the [standard upgrade](#standard-upgrade), then after you've completed the upgrade, return to the [post-upgrade configuration changes instructions](#v0.4.0-post-upgrade-configuration-changes) to update your `GatewayClass`, `Gateway` and `HTTPRoute` resources from v1alpha2 to v1beta1, and replace your `ReferencePolicy` resources with `ReferenceGrant`.
+
+<a name="v0.4.0-post-upgrade-configuration-changes"/>
+
+### Post-Upgrade Configuration Changes
+
+#### Requirements
+
+- Consul API Gateway should be running version v0.4.0.
+- Consul Helm chart should be v0.47.0 or later.
+- You should have the ability to run `kubectl` CLI commands.
+- `kubectl` should be configured to point to the cluster containing the installation you are upgrading.
+- You should have the following permission rights on your Kubernetes cluster:
+  - `Gateway.read`
+  - `ReferenceGrant.create` (Added in Consul Helm chart v0.47.0)
+  - `ReferencePolicy.delete`
+
+#### Procedure
+
+1. Verify the current version of the `consul-api-gateway-controller` `Deployment`:
+
+  ```shell-session
+  $ kubectl get deployment --namespace consul consul-api-gateway-controller --output=jsonpath="{@.spec.template.spec.containers[?(@.name=='api-gateway-controller')].image}"
+  ```
+
+  You should receive a response similar to the following:
+
+  ```log
+  "hashicorp/consul-api-gateway:0.4.0"
+  ```
+
+  <a name="referencegrant"/>
+
+1. Issue the following command to get all `ReferencePolicy` resources across all namespaces. If you have any active `ReferencePolicy` resources, you will receive output similar to the response below.
+
+  ```shell-session
+  $ kubectl get referencepolicy --all-namespaces
+  Warning: ReferencePolicy has been renamed to ReferenceGrant. ReferencePolicy will be removed in v0.6.0 in favor of the identical ReferenceGrant resource.
+  NAMESPACE   NAME
+  default     example-reference-policy
+  ```
+
+  If your output is empty, proceed to upgrading your `GatewayClass`, `Gateway` and `HTTPRoute` resources to [v1beta1](#v1beta1-gatewayclass-gateway-httproute).
+
+1. For each `ReferencePolicy`, edit the source YAML file, updating to `kind: ReferenceGrant` and optionally updating the `metadata.name` field or filename if they previously included the term "policy", as in the example below. Note that because the `kind` field is being updated, it is not possible to use `kubectl edit` to edit the remote state directly.
```

Comment:
```suggestion
1. For each `ReferencePolicy` in the YAML file, change the `kind` field to `ReferenceGrant`. You can optionally update the `metadata.name` field or filename if they include the term "policy". In the following example, the `kind` and `metadata.name` fields have been changed to the new resource. Note that updating the `kind` field prevents you from using the `kubectl edit` command to edit the remote state directly. 
```
Trying to improve the flow; better?

Reply:
I think specifying `source YAML` (as preserved in the suggestion below for v1beta1 upgrades) is relevant to contrast with editing the live resource on the Kubernetes server - will add that word back and merge this.